### DISCUSSION TITLE
Autoroute Path not being correctly generated when ContentItem is created via code.

### DISCUSSION THREAD

**Discussion by XpandIT-RFID** [OrchardCMS/OrchardCore#10204] [Q&A] [answered]
Hi, I have the following code in my project:

```
var contentItem = await _contentManager.NewAsync("Testimony");
contentItem.Content.TitlePart.Title = "Test"
await _contentManager.CreateAsync(contentItem, VersionOptions.Draft);
await _contentManager.UpdateAsync(contentItem);
``` 
*simplified version of the code 👍

The AutoroutePart for this "Testimony" Content is defined like this: 
![image](https://user-images.githubusercontent.com/83280308/131871783-11d91e64-0cf2-419a-b5bb-b4e4a1445a49.png)

But when I create this content item via code, the permalink comes as "testimonials/-1" instead of "testimonials/Test".

Can anyone help me with this? I have been looking at the AutoroutePart Module but I only got more confused XD.

cc: @Skrypt 

OC Version : 1.0.0

**Comment by Skrypt** [comment, marked as answer]
You need to use `contentItem.DisplayText = "Test"` else you could try to change your Liquid to use `{{ ContentItem.Content.TitlePart.Title | slugify }}`

DisplayText sets it's value from the TitlePart.Title from a handler and it seems that the value is not affected on the ContentItem yet when it gets parsed by this Liquid template. There is probably a small issue there but it should work by doing what I described above.

**Reply by XpandIT-RFID** [reply]
I was doing `ContentItem.Content.DisplayText` instead of `ContentItem.DisplayText` on my code for some weird reason... I didn't even put it in the sample above because of it being generated in the handlers, but yeah, there is that issue you mentioned 👍🏼 

Thanks for the help again @Skrypt 🥇
